### PR TITLE
Link component for the download section

### PR DESCRIPTION
**Pull request by alex6061**

### CODE REVIEW

File: js/components/widgets/Link.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @typedef {Object}
+ * @param {Object} data 
+ * @param {string} parrentClassName 
+ * @param {string} description 
+ * @param {string} url 
+ * @returns {string} HTML or empty 
+ */
+
+export const Link = (url, description, parrentClassName) => {
+  if(!Link) return '';
```
Comment (nagorapro) [inline js/components/widgets/Link.js:12]: What is this line for?

File: js/components/widgets/Link.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  const { description, url } = data;
```
Comment (nagorapro) [inline js/components/widgets/Link.js:14]: Where did the "data" identifier come from?

File: js/components/widgets/Link.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  const currentClassName = parrentClassName ?
+   `${parrentClassName}__link` :
+    `link`;
+
+  return `
+    <a
+      class="${currentClassName}"
+      src="${Link.url}"
```
Comment (nagorapro) [inline js/components/widgets/Link.js:23]: Error

File: js/components/widgets/Link.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      alt="${Link.description}"
```
Comment (nagorapro) [inline js/components/widgets/Link.js:24]: Error

File: js/components/widgets/Link.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @function Link
+ * @typedef {Object}
+ * @param {Object} data
```
Comment (nagorapro) [inline js/components/widgets/Link.js:4]: Bad

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Link.js (hunk 1 above)
Comment (alex6061) [reply js/components/widgets/Link.js:12]: > What is this line for?

line 12 says that if the function does not take a reference, then the function should not continue to perform its actions, but should return an empty string

Comment (nagorapro) [reply js/components/widgets/Link.js:12]: Link is your function name
You don't have the Link parameter!!!

File: js/components/widgets/Link.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ * @typedef {Object}
+ * @property {Data} data
+ * @param {Object} url 
+ * @param {string} parrentClassName 
+ * @param {string} description 
+ * @returns {string} HTML or empty 
+ */
+
+export const Link = (url, description, parrentClassName) => {
+  if(!data) return '';
```
Comment (nagorapro) [inline js/components/widgets/Link.js:12]: if (!Object.keys(data).length === 2) return '';

File: js/components/widgets/Link.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+  const { description, url, link } = data;
```
Comment (nagorapro) [inline js/components/widgets/Link.js:14]: const { description, url } = data;

File: js/components/widgets/Link.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+  const currentClassName = parrentClassName 
+    ? `${parrentClassName}__link` 
+    : `link`;
+
+  return `
+    <a
+      class="${currentClassName}"
+      src="${link.url}"
```
Comment (nagorapro) [inline js/components/widgets/Link.js:23]: href="${url}"

File: js/components/widgets/Link.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      alt="${link.description}"
```
Comment (nagorapro) [inline js/components/widgets/Link.js:24]: Delete line

File: js/components/widgets/Link.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    >
+      ${Link.type === 'google' && IconGoogle()}
+      ${Link.type === 'apple' && IconApple()}
```
Comment (nagorapro) [inline js/components/widgets/Link.js:27]: ${type === 'google' ? IconGoogle() : IconApple()}

File: js/components/widgets/Link.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @function Link
+ * @typedef {Object}
+ * @property {Data} data
+ * @param {Object} url 
+ * @param {string} parrentClassName 
+ * @param {string} description 
+ * @returns {string} HTML or empty
```
Comment (nagorapro) [inline js/components/widgets/Link.js:8]: `
 * @function Link
 * @typedef {Object} Data
 * @property {string} type
 * @property {string} url
 * @param {Data} data
 * @param {string} parrentClassName 
 * @returns {string} HTML or empty
 `

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Link.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ * @typedef {Object} Data
+ * @property {string} type
+ * @property {string} url
+ * @param {Data} data
+ * @param {string} parrentClassName
+ * @returns {string} HTML or empty
+ */
+
+export const Link = (data, parrentClassName) => {
+  if (!Object.keys(data).length === 2) return "";
```
Comment (nagorapro) [inline js/components/widgets/Link.js:12]: if (Object.keys(data).length !== 2) return '';

File: js/components/widgets/Link.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  const { type, link } = data;
```
Comment (nagorapro) [inline js/components/widgets/Link.js:13]: Why link?
* @property {string} url

File: js/components/widgets/Link.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  const currentClassName = parrentClassName
+    ? `${parrentClassName}__link`
+    : `link`;
+
+  return `
+    <a
+      class="${currentClassName}"
+      src="${url}"
+    >
+      ${type === "google" ? IconGoogle() : IconApple()}
```
Comment (nagorapro) [inline js/components/widgets/Link.js:24]: Use ' '

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Link.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @function Link
+ * @typedef {Object} Data
+ * @property {string} type
+ * @property {string} url
+ * @param {Data} data
+ * @param {string} parrentClassName
+ * @returns {string} HTML or empty
+ */
```
Comment (nagorapro) [inline js/components/widgets/Link.js:9]: Use 2 JSDoc

File: js/components/widgets/Image.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+  if (!Object.keys(data).length !== 2) return '';
+
+  const currentClassName = parrentClassName 
+    ? `${parrentClassName}__image` 
+    : `image`;
+
+    return `
+      <img
+        class="${currentClassName}"
+        src="${url}"
```
Comment (nagorapro) [inline js/components/widgets/Image.js:21]: Where is the url from?

File: js/components/widgets/Image.js [hunk 17]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @function Image
+ * @typedef {Object} Data
+ * @property {string} type
+ * @property {url} type
+ * @param {Data} data
+ * @param {string} parrentClassName
+ * @returns {string} HTML or empty
+ */
```
Comment (nagorapro) [inline js/components/widgets/Image.js:9]: Use 2 JSDoc

Why is this file in the Link-component-for-download-section branch?

**Review by nagorapro — COMMENTED** [review]
Need fix

File: js/components/widgets/Link.js [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+ * @function Link
+ * @param {Data} data
+ * @param {string} parrentClassName
+ * @returns {string} HTML or empty
+ */
+
+export const Link = (data, parrentClassName) => {
+  if (Object.keys(data).length !== 2) return '';
+
+  const { type, url } = data.links;
```
Comment (nagorapro) [inline js/components/widgets/Link.js:17]: Bad

const { type, url } = data;

**Review by nagorapro — COMMENTED** [review]
Need fix

**Pull request merged**